src/core/Database.cpp

```diff
@@ -241,27 +242,30 @@ bool Database::save(const QString& filePath, QString* error, bool atomic, bool b
 
             // Delete the original db and move the temp file in place
             QFile::remove(filePath);
-#ifdef Q_OS_LINUX
-            // workaround to make this workaround work, see: https://bugreports.qt.io/browse/QTBUG-64008
-            if (tempFile.copy(filePath)) {
-                // successfully saved database file
-                return true;
-            }
-#else
-            if (tempFile.rename(filePath)) {
-                // successfully saved database file
+
+            // Note: call into the QFile rename instead of QTemporaryFile
+            // due to an undocumented difference in how the function handles
+            // errors. This prevents errors when saving across file systems.
+            if (tempFile.QFile::rename(filePath)) {
+                // successfully saved the database
                 tempFile.setAutoRemove(false);
                 setFilePath(filePath);
                 return true;
+            } else {
+                // restore the database from the backup
+                if (backup) {
+                    restoreDatabase(filePath);
```

Comment:
This needs a check if database backups are enabled. If not, we can only disable autoremove and display the name of the temporary savefile instead.